### PR TITLE
Allow Collapsing Selected Components, Add Hotkeys for Collapsing Components

### PR DESCRIPTION
**Pull request by Ghrehh**
## Description
- Allow collapsing components, even those that the user is currently selecting. If the user selects a component contained within a collapsed parent via the app preview, that component will be temporarily visible in the tree.
- Add hotkeys and help menu options for collapsing, expanding and recursively collapsing and expanding components.
- Persists collapsed components in session storage, so that collapsed components are at least temporarily maintained if the user for instance navigates between the data and design sections.

## Addresses
- https://linear.app/budibase/issue/BUDI-7675/design-section-component-tree-dont-expand-all-components

## Feature branch env
[Feature Branch Link](http://fb-fix-component-tree-collapsing-2.fb.qa.budibase.net)

### CODE REVIEW

**Review by deanhannigan — CHANGES_REQUESTED** [review]
LGTM! Minor issue with the tree node selection but the feature works nicely when navigating around the builder 👍

File: packages/builder/src/pages/builder/app/[application]/design/[screenId]/_components/ComponentList/ComponentDropdownMenu.svelte [hunk 1]
```diff
@@ -85,6 +86,39 @@
   >
     Paste
   </MenuItem>
+
+  {#if component._children.length > 0}
```
[the remark this answers is not in the document]
Comment (Ghrehh) [reply packages/builder/src/pages/builder/app/[application]/design/[screenId]/_components/ComponentList/ComponentDropdownMenu.svelte]: Good catch, should be fixed now.

**Review by deanhannigan — APPROVED** [review]
LGTM!

**Comment by Ghrehh** [comment]
Note: We had an issue where the order of the `:fix` version of our lint script was in a different order than the non-fix version, which lead to a weird intermittent failure here, should be fixed now.

**Pull request merged**